src/core/ConsoleVariable.h

```diff
@@ -4,15 +4,73 @@
 #include <nlohmann/json.hpp>
 #include <stdint.h>
 #include <memory>
-#include <map>
 #include <string>
+#include <type_traits>
+#include <unordered_map>
+#include <variant>
 
 namespace Ship {
 typedef enum class ConsoleVariableType { Integer, Float, String, Color, Color24 } ConsoleVariableType;
 typedef union CVarValue {
 
 } CVarValue;
 
+class CVarInterface {
+  public:
+    using VauleType = std::variant<int32_t, float, std::string, Color_RGBA8, Color_RGB8>;
```

Comment:
In lieu of an enum to store the type, I opted for `std::variant`. This makes it more type safe and allows everything to be templatized.